tests/WebAppCallsWebApiCallsGraph/Client/Services/TodoListService.cs

```diff
@@ -2,158 +2,96 @@
+            this MicrosoftIdentityAppCallsWebApisAuthenticationBuilder builder, 
+            IConfiguration configuration)
         {
+            builder.AddDownstreamApiService(TodoListService.ServiceName, configuration);
             // https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests
-            services.AddHttpClient<ITodoListService, TodoListService>();
+            builder.Services.AddHttpClient<ITodoListService, TodoListService>();
+            return builder;
         }
     }
 
     /// <summary></summary>
     /// <seealso cref="TodoListClient.Services.ITodoListService" />
     public class TodoListService : ITodoListService
     {
-        private readonly IHttpContextAccessor _contextAccessor;
-        private readonly HttpClient _httpClient;
-        private readonly string _TodoListScope = string.Empty;
-        private readonly string _TodoListBaseAddress = string.Empty;
-        private readonly ITokenAcquisition _tokenAcquisition;
-        private readonly JsonSerializerOptions _jsonOptions = new JsonSerializerOptions
-        {
-            PropertyNameCaseInsensitive = true
-        };
+        private readonly IDownstreamWebApi _downstreamWebApi;
+        public const string ServiceName = "TodoList";
 
-        public TodoListService(ITokenAcquisition tokenAcquisition, HttpClient httpClient, IConfiguration configuration, IHttpContextAccessor contextAccessor)
+        public TodoListService(IDownstreamWebApi downstreamWebApi)
         {
-            _httpClient = httpClient;
-            _tokenAcquisition = tokenAcquisition;
-            _contextAccessor = contextAccessor;
-            _TodoListScope = configuration["TodoList:TodoListScope"];
-            _TodoListBaseAddress = configuration["TodoList:TodoListBaseAddress"];
+            _downstreamWebApi = downstreamWebApi;
         }
 
         public async Task<Todo> AddAsync(Todo todo)
         {
-            var httpRequestMessage = await PrepareAuthenticatedClient(
-              $"{ _TodoListBaseAddress}/api/todolist",
-              HttpMethod.Post);
-
-            var jsonRequest = JsonSerializer.Serialize(todo);
-            var jsoncontent = new StringContent(jsonRequest, Encoding.UTF8, "application/json");
-            httpRequestMessage.Content = jsoncontent;
-
-            var response = await _httpClient.SendAsync(httpRequestMessage);
-
-            if (response.StatusCode == HttpStatusCode.OK)
-            {
-                var content = await response.Content.ReadAsStringAsync();
-                todo = JsonSerializer.Deserialize<Todo>(content, _jsonOptions);
-
-                return todo;
-            }
-
-            throw new HttpRequestException($"Invalid status code in the HttpResponseMessage: {response.StatusCode}.");
+            return await _downstreamWebApi.CallWebApiForUserAsync<Todo, Todo>(
```

Comment:
use strongly typed wrappers and express just what is different. The code is way simpler and no copy/paste any longer.